controllers/egressipam/azure/azure.go

```diff
@@ -452,6 +457,9 @@ func (i *AzureInfra) addNeededAzureAssignedIPs(rc *reconcilecontext.ReconcileCon
 					}
 				}
 			}
+			if attachInternalLoadBalancerPool {
```

Comment:
please move the call to the function `shouldAttachInternalLoadBalancerPool` here instead of using a temporary variable. Even better would be to use a case statement here as I think there will be more cases in the future. Anyway the if will suffice for now.

Reply:
Agreed on the optimization. I made some refactoring in the development process and there is no longer a need to separate out. Implementing as suggested